ui/core/package.json

```diff
@@ -33,7 +33,8 @@
     "lodash": "^4.17.21",
     "mathjs": "^10.6.4",
     "numbro": "^2.3.6",
-    "zod": "^3.21.4"
+    "zod": "^3.21.4",
+    "use-query-params": "^2.1.2"
```

Comment:
👋 Quickly skimming, but the ProjectStoreProvider move to core makes me nervous, wouldn’t it bring other dependencies too like react-query? Really core is meant to be the lowest level, most minimal package, mostly just for shared types and utils, not really providers, data fetching, or query param logic (really I had always wanted to refactor and make use-query-params completely optional even in plugin-system) 

What we probably need is some type of AppMetadataProvider that could even be decoupled from the concept of “projects”
That will take some trial and error to get around circular dependencies but even if that had to live in plugin-system, I think it’d be better than coupling core to the app

Reply:
Yeah!
I won't lie I feel the same.
But I didn't get a better idea.
I'm not sure if we will be able build such a generic metadata provider but we can give a try

Reply:
Yeah maybe it’s something we aspire to in the future, if this PR works for Red Hat’s use case it’s probably fine!

Would moving ProjectStoreProvider somewhere besides core be possible though (maybe plugin-system?), the other reason is I don’t want to indirectly make the components package depend on react-query, use-query-params, etc (this [old dependency diagram](https://github.com/perses/perses/blob/v0.22.0/ui/README.md#library-architecture-diagram) might be helpful to reference too)

Reply:
Let's move it to plugin-system for now and we can feel if we are going to have more use cases like that and see what we should do.

Wdyt?